client/signup/utils.js

```diff
@@ -93,6 +95,27 @@ function getNextStepName( flowName, currentStepName ) {
 	return flow.steps[ indexOf( flow.steps, currentStepName ) + 1 ];
 }
 
+function getValueFromProgressStore( { signupProgressStore, stepName, fieldName } ) {
+	if ( ! signupProgressStore ) {
+		return null;
+	}
+	const siteStepProgress = find(
+		signupProgressStore,
+		step => step.stepName === stepName
+	);
+	return siteStepProgress ? siteStepProgress[fieldName] : null;
+}
+
+function mergeFormWithValue( { form, fieldName, fieldValue} ) {
+	let fieldToMergeValue = formState.getFieldValue( form, fieldName );
+	if ( fieldToMergeValue == null ) {
```

Comment:
`fieldToMergeValue` can be declared with `const` because we only assign to it once, but you may as well just omit it with:

```js
if ( ! formState.getFieldValue( form, fieldName ) ) {
```

This will merge for all falsy values (including an empty string), which is different than the behavior above (which will merge for `undefined` and `null`). I think either behavior is probably fine, but if we want to check for null, we should do a strict `===` comparison so that we know we're merging fields that have been initialized and set to `null`.